src/algebra/non_associative/basic.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import algebra.algebra.basic
+
+/-!
+# Non-associative, non-unital algebras
+
+This files defines non-associative, non-unital algebras together with their morphisms and basic
+lemmas to make them usable.
+
+## Main definitions
+
+  * `na_semiring`
+  * `na_algebra`
+  * `na_algebra_hom`
+
+## Tags
+
+non-associative, non-unital, algebra
+-/
+
+universes u₁ u₂ u₃ u₄
+
+variables (R : Type u₁) (A : Type u₂) (B : Type u₃) (C : Type u₄) [semiring R]
```

Comment:
Note that I don't assume `R` is commutative. Perhaps I should?

For unital algebras we always assume commutativity since any action factors through the commutative quotient but I don't see that this is the case here.